ci/Dockerfile.newrelic.lambda

```diff
@@ -1,6 +1,9 @@
+# Stage 1: Copy New Relic Lambda Layer for Python 3.12
+FROM public.ecr.aws/newrelic-lambda-layers-for-docker/newrelic-lambda-layers-python:312 AS layer
+
+# Stage 2: Build application
 FROM python:3.12-slim@sha256:31a416db24bd8ade7dac5fd5999ba6c234d7fa79d4add8781e95f41b187f4c9a
 
-ENV PYTHONPATH "${PYTHONPATH}:/opt/python/lib/python3.12/site-packages"
```

Comment:
Was this necessary for poetry? was this tested locally with the removal of the env var? 👀